src/zkscript/merkle_trees/merkle_tree.py

```diff
@@ -0,0 +1,198 @@
+from tx_engine import Script
+
+
+class MerkleTree:
+    """Class representing a Merkle Tree.
+
+    This class takes a root hash and a specified hash function to create the Merkle Tree
+    structure.
+
+    Attributes:
+        root (str): The root hash of the Merkle Tree. Must be a valid hexadecimal value.
+        hash_function (str): The hash function used in the Merkle Tree. This must be a valid opcode hash function or a
+            combination of valid opcode hash functions.
+        depth (int): The depth (number of levels) of the Merkle tree.
+
+    Methods:
+        __init__(root: str, hash_function: str): Initializes the Merkle Tree. An assertion is
+            performed to ensure the hash function is valid and the root is hexadecimal.
+
+    """
+
+    def __init__(self, root: str, hash_function: str, depth: int):
+        assert all(c in "0123456789abcdefABCDEF" for c in root), f"{root} is not a valid hexadecimal string."
+
+        assert set(hash_function.split(" ")).issubset(
+            {"OP_RIPEMD160", "OP_SHA1", "OP_SHA256", "OP_HASH160", "OP_HASH256"}
+        ), f"{hash_function} is not a valid hash function."
+
+        assert depth > 0
+
+        self.root = root
+        self.hash_function = hash_function
+        self.depth = depth
+
+    def locking_merkle_proof_with_bit_flags(
+        self,
+        is_equal_verify: bool = False,
+    ) -> Script:
+        """Generate a locking script to verify a Merkle path.
+
+        The locking script returned by this function checks the validity of a Merkle path. For a root r, the
+        Merkle path for an element d is defined as a sequence:
+            (d, "", "") (h_1, aux_1, bit_1) ... (h_{self.depth-1}, aux_{self.depth-1}, bit_{self.depth-1})
+        So that:
+            - h_1 = sef.hash(d)
+            - h_i = self.hash(h_{i-1} || aux_{i-1}) if bit_{i-1} == 0 else self.hash(aux_{i-1} || h_{i-1})
+            - r = self.hash(h_{self.depth-1} || aux_{self.depth-1}) if bit_{self.depth-1} == 0
+                    else self.hash(aux_{self.depth-1} || h_{self.depth-1})
+
+        Args:
+            is_equal_verify: a boolean variable to choose between `OP_EQUAL` or `OP_EQUALVERIFY` as final opcode in the
+                script. Default to `False`.
+
+        Returns:
+            A locking script verifying a Merkle path.
+
+        Note:
+            - `self.hash_function` must be a combination of valid Bitcoin Script hash functions (e.g., `OP_SHA256`).
```

Comment:
Why a combination?

Reply:
Maybe it would make more sense to have it as a list then

Reply:
you mean write list instead of combination in the description? I am fine with changing combination to something clearer, if it helps, maybe I would use sequence rather than list, though

Do you mean passing a list instead of a string? I don't agree for two reasons: 
    1. if we have list instead of strings, most of the time is going just to be a clumsy ["op_sha256"] argument to the function. 
    2. Strings have a natural order, while lists have two reasonable orders: if I pass "op_sha op_hash", I expect to find precisely that string in the locking script, if I passed ["op_sha", "op_hash"] I would have to check if which hash function is performed first

Reply:
I rewrote the description a bit, should be a bit more informative